streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreBuilderWrapper.java

```diff
@@ -0,0 +1,119 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.apache.kafka.streams.state.internals.SessionStoreBuilder;
+import org.apache.kafka.streams.state.internals.TimestampedWindowStoreBuilder;
+import org.apache.kafka.streams.state.internals.VersionedKeyValueStoreBuilder;
+import org.apache.kafka.streams.state.internals.WindowStoreBuilder;
+
+public class StoreBuilderWrapper<S extends StateStore> implements StoreFactory<S> {
+
+    private final StoreBuilder<S> builder;
+    private final Set<String> users = new HashSet<>();
+
+    public StoreBuilderWrapper(final StoreBuilder<S> builder) {
+        this.builder = builder;
+    }
+
+    @Override
+    public S build() {
+        return builder.build();
+    }
+
+    @Override
+    public long retentionPeriod() {
+        if (builder instanceof WindowStoreBuilder) {
+            return ((WindowStoreBuilder<?, ?>) builder).retentionPeriod();
+        } else if (builder instanceof TimestampedWindowStoreBuilder) {
+            return ((TimestampedWindowStoreBuilder<?, ?>) builder).retentionPeriod();
+        } else if (builder instanceof SessionStoreBuilder) {
+            return ((SessionStoreBuilder<?, ?>) builder).retentionPeriod();
+        } else {
+            throw new IllegalStateException(
+                    "retentionPeriod is not supported when not a window store");
+        }
+    }
+
+    @Override
+    public long historyRetention() {
+        if (builder instanceof VersionedKeyValueStoreBuilder) {
+            return ((VersionedKeyValueStoreBuilder<?, ?>) builder).historyRetention();
+        } else {
+            throw new IllegalStateException(
+                    "historyRetention is not supported when not a versioned store");
+        }
+    }
+
+    @Override
+    public Set<String> users() {
+        return users;
+    }
+
+    @Override
+    public boolean loggingEnabled() {
+        return builder.loggingEnabled();
+    }
+
+    @Override
+    public String name() {
+        return builder.name();
+    }
+
+    @Override
+    public boolean isWindowStore() {
+        return builder instanceof WindowStoreBuilder
+                || builder instanceof TimestampedWindowStoreBuilder
+                || builder instanceof SessionStoreBuilder;
+    }
+
+    @Override
+    public boolean isVersionedStore() {
+        return builder instanceof VersionedKeyValueStoreBuilder;
+    }
+
+    // Apparently Java strips the generics from this method because we're using the raw type for builder,
+    // even though this method doesn't use builder's (missing) type parameter. Our usage seems obviously
+    // correct, though, hence the suppression.
```

Comment:
What suppression...?I feel like I recognize this comment, did it become detached from its original annotation?

Reply:
I somehow missed that this class was just extracted from InternalTopologyBuilder on my first pass, and just deleted a bunch of comments on this stuff...but I'm leaving up this one, because I think you can take this opportunity to at least remove the detached code comment

Reply:
sounds good, I had no idea what the comment meant but didn't want to remove it.